usecases/tasks_handler.go

```diff
@@ -0,0 +1,251 @@
+package usecases
+
+import (
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/blcvn/corev3-libs/flogging"
+	contractpb "github.com/blcvn/corev3-protos/v2/types/smartcontract"
+	"github.com/blcvn/corev4-explorer/common"
+	"github.com/blcvn/corev4-explorer/entities"
+)
+
+type dbRepo interface {
+	CalculateAccountDelta(taskId int64) error
+	CalculateAccountBalance(taskId int64) error
+	SaveTransformData(taskId int64, onchainTransactions []*entities.OnchainTransaction, accounts []*entities.Account, accountTxsList []*entities.AccountTx) error
+	QueryRawTxsByTaskTransformID(taskId int64) ([]*entities.RawTransaction, error)
+	// CreateTask(task *entities.Task) error
+	// GetAllTaskByStatus(taskType int32, taskStatus int32) ([]*entities.Task, error)
+	// UpdateTasksStatus(taskIds []int64, taskStatus int32) error
+	// UpdateTaskStatus(taskId int64, taskStatus int32) error
+	// FindOneById(taskId int64) (*entities.Task, error)
+	// GetRawTransactionByBlockNumber(blockNumber uint64) ([]*entities.RawTransaction, error)
+}
+
+type taskHandler struct {
+	dbRepo dbRepo
+	logger *flogging.FabricLogger
+}
+
+func NewTaskHandler(dbRepo dbRepo) *taskHandler {
+	return &taskHandler{
+		dbRepo: dbRepo,
+		logger: flogging.MustGetLogger("corev4-explorer.usecase.task.handler"),
+	}
+}
+
+func (t *taskHandler) PerformTasks(taskId int64, tasksType int32) common.BaseError {
+	switch tasksType {
+	case int32(entities.TaskTransform):
+		return t.transformData(taskId)
+	case int32(entities.TaskBalance):
+		return t.calculateAccountDelta(taskId)
+	default:
+		return common.NewUnknownError(errors.New("unknown task"))
+	}
+}
+
+func (t *taskHandler) transformData(taskId int64) common.BaseError {
+	t.logger.Infof("Transform task %d start", taskId)
+	rawTxs, err := t.dbRepo.QueryRawTxsByTaskTransformID(taskId)
+	if err != nil {
+		return common.NewError(common.ERROR_INTERNAL, err)
+	}
+
+	var onchainTxList []*entities.OnchainTransaction
+
+	var accountList []*entities.Account
+
+	var accountTxs []*entities.AccountTx
+
+	for _, rawTransaction := range rawTxs {
+
+		if strings.HasPrefix(rawTransaction.ChaincodeProposalInput, "Transaction.") {
+			newTransactions, newAccountTxs, err := t.rawTxToTransactions(rawTransaction, taskId)
+			if err != nil {
+				return err
+			}
+			onchainTxList = append(onchainTxList, newTransactions...)
+			accountTxs = append(accountTxs, newAccountTxs...)
+
+		}
+
+		if strings.HasPrefix(rawTransaction.ChaincodeProposalInput, "Account.") {
+			newAccounts, err := t.rawTxToAccounts(rawTransaction, taskId)
+			if err != nil {
+				return err
+			}
+			accountList = append(accountList, newAccounts...)
+
+		}
+
+	}
+	err = t.dbRepo.SaveTransformData(taskId, onchainTxList, accountList, accountTxs)
+	if err != nil {
+		return common.NewError(common.ERROR_INTERNAL, err)
+	}
+	return nil
```

Comment:
Implement the `calculateAccountDelta` function to perform the intended logic for calculating account deltas. Currently, it returns `nil` which might be a bug or placeholder. [important]